src/TreeView/TreeView.tsx

```diff
@@ -257,8 +262,15 @@ const Root: React.FC<TreeViewProps> = ({'aria-label': ariaLabel, 'aria-labelledb
     setAriaLiveMessage(message)
   }, [])
 
+  const expandedStateCache = React.useRef(new Map())
+
```

Comment:
Small suggestion to avoid creating a new Map each render:

```suggestion
  const expandedStateCache = React.useRef(null)

  if (expandedStateCache.current === null) {
    expandedStateCache.current = new Map();
  }
```